### PR TITLE
Use `test` sub-namespace instead of `celeritas_test`

### PR DESCRIPTION
**Pull request by sethrj**
Change namespace on lots of tests to `celeritas::test`. This allows them to use everything in the `celeritas` namespace.

~The only really confusing bit is that tests for the `celeritas::detail` are in `celeritas::detail::test`; and details for celeritas testing are in `celeritas::test::detail`... and then detail tends to become ambiguous.~

I changed all the "low-level testing helper details" to be in the `testdetail` namespace (macros, main) and scoped the file names at the top level (so now it's `TestMacros.hh`) and a little easier to read. I've moved helper classes out of the `detail` subdirectory: the only `detail` directories inside `test` should now be *tests of details* not *details of tests*. (heads up to @whokion ).

### CODE REVIEW

**Comment by pcanal** [comment]
> celeritas::test::detail

Why does a test need a detail sub-namespace?  Isn't detail's intent to signal that even-though the content is in published headers, it is not intended to be using by "end-users" (i.e. subject to change without notice).

Can't we either remove it or rename it to a name closer to its current semantic?

**Comment by sethrj** [comment]
@pcanal I'm happy for suggestions? My constainst so far have been:
- `celeritas_test.hh` should provide everything a test needs, including the `main` definition
- `something/Test.hh` should allow you to define a new test inheriting from the celeritas (and google test)
- `something/macros.hh` should provide the macros needed to add testing to a helper class outside of  a main test

so mainly the macros and test harness have been included transitively (and the functions they call have been "details"). Any thoughts?

**Comment by pcanal** [comment]
So if I understand correctly, the content of current `celeritas::test::detail` are implementation details of the test framework/harness that are not supposed to be called directly by the actual test.  (if the only goals is to disambiguate for the compiler your could use `details` :) )  More seriously you could use a 'synonym' for details, like "impl[ementation]", "internal", etc.

**Comment by pcanal** [comment]
Note: The high level discussion is very likely enough but the ratio of clang-format to spelling changes is too high to be able to "see" the effect on the actual code.

**Review by amandalund — APPROVED** [review]
Changes look good @sethrj, definitely an improvement!

File: test/detail/Macros.test.cc [hunk 1]
```diff
@@ -13,14 +13,20 @@
...
 constexpr double inf = std::numeric_limits<double>::infinity();
 
+namespace celeritas
+{
+namespace test
+{
+namespace detail
+{
+namespace test
+{
```
[the remark this answers is not in the document]
Comment (sethrj) [reply test/detail/Macros.test.cc]: 🤪 this has been fixed ;)

File: test/celeritas/em/CombinedBrem.test.cc [hunk 2]
```diff
@@ -71,18 +64,18 @@ class CombinedBremTest : public celeritas_test::InteractorHostTestBase
         std::vector<celeritas::ImportProcess> imported{
             {11,
              22,
-             celeritas::ImportProcessType::electromagnetic,
-             celeritas::ImportProcessClass::e_brems,
+             ImportProcessType::electromagnetic,
+             ImportProcessClass::e_brems,
              {celeritas::ImportModelClass::e_brems_sb,
```
[the remark this answers is not in the document]
Comment (sethrj) [reply test/celeritas/em/CombinedBrem.test.cc]: Took care of it :) thanks for checking it out.

**Pull request merged**